### DISCUSSION TITLE
Isn't query alternation commutative?

### DISCUSSION THREAD

**Discussion by champignoom** [tree-sitter/tree-sitter#2275] [Q&A]
I have a document that is parsed into `(p (node1) (q (node2)))`.

This query matches both node1 and node2:
```
[ (nodeFoo) @x [ (node1) @x (node2) @x ] ]
```

But this only matches node1:
```
[ [ (node1) @x (node2) @x ] (nodeFoo) @x ]
```

How is this happening? Isn't alternation commutative for disjoint matches?

**Comment by ahlinc** [comment]
Could you give a reproducible real example? I'd like to look on the real behavior first on a debug version of tree-sitter.
